include/crm/common/strings_internal.h

```diff
@@ -53,6 +53,21 @@ GHashTable *pcmk__strikey_table(GDestroyNotify key_destroy_func,
                                 GDestroyNotify value_destroy_func);
 GHashTable *pcmk__str_table_dup(GHashTable *old_table);
 
+/*!
+ * \internal
+ * \brief Get a string value with a default if NULL
+ *
+ * \param[in] s              String to return if non-NULL
+ * \param[in] default_value  String (or NULL) to return if \p s is NULL
+ *
+ * \return \p s if \p s is non-NULL, otherwise \p default_value
+ */
+static inline const char *
+pcmk__s(const char *s, const char *default_value)
+{
+    return (s == NULL)? default_value : s;
+}
+
 /*!
```

Comment:
I think that would make the calls long. This needs to be as short as possible to fit in initializations and log messages on one line. It's rare to have a chain of defaults anyway.

Reply:
above should have been
s/distinction between and would/distinction between null-pointer and empty-string would/

Reply:
The idea of going with a va-list would actually have been to make it even shorter - I mean the source-line.
Regarding size of binary-code it would depend on how capable optimization is here or in how far it can be
hinted. As the macros meanwhile don't come from a library but the compiler there might be a chance.
Didn't play with it nor found any info right away.